src/helpers.ts

```diff
@@ -88,3 +88,19 @@ export function createDefaultClient(
 
   return { client, publicKey, address }
 }
+
+export function createDefaultEthSignClient(
+  dappchainKey: string,
+  dappchainEndpoint: string,
+  chainId: string,
+  wallet: ethers.Signer
+): { client: Client; publicKey: Uint8Array; address: Address } {
+  const defaultClientObj = createDefaultClient(dappchainKey, dappchainEndpoint, chainId)
+
+  defaultClientObj.client.txMiddleware = [
+    new NonceTxMiddleware(defaultClientObj.publicKey, defaultClientObj.client),
```

Comment:
Should pass the `eth:0x...` address to the `NonceTxMiddleware`, not the DAppChain public key.